examples/case_studies/BART_quantile_regression.myst.md

```diff
@@ -0,0 +1,171 @@
+# Quantile Regression with BART
+:::{post} Jan 25, 2023
+:tags: BART, non-parametric, quantile, regression 
+:category: intermediate, explanation
+:author: Osvaldo Martin
+:::
+
+```{code-cell} ipython3
+from pathlib import Path
+
+import arviz as az
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import pymc as pm
+import pymc_bart as pmb
+
+from scipy import stats
+
+print(f"Running on PyMC v{pm.__version__}")
+```
+
+```{code-cell} ipython3
+%config InlineBackend.figure_format = "retina"
+RANDOM_SEED = 5781
+np.random.seed(RANDOM_SEED)
+az.style.use("arviz-darkgrid")
+```
+
+Usually when doing regression we model the conditional mean of some distribution. Common cases are a Normal distribution for continuous unbounded responses, a Poisson distribution for count data, etc.
+
+Quantile regression, instead estimates a conditional quantile of the response variable. If the quantile is 0.5, then we will be estimating the median (instead of the mean), this could be useful as a way of performing robust regression, in a similar fashion as using a Student-t distribution instead of a Normal. But for some problems we actually care of the behavior of the response away from the mean (or median). For example, in medical research, pathologies or potential health risks occur at high or low quantiles, for instance, overweight and underweight. In some other fields like ecology, quantile regression is justified due to the existence of complex interactions between variables, where the effect of one variable on another is different for different ranges of the variable.
+
++++
+
+## Asymmetric Laplace distribution
+
+At first it could be weird to think which distribution we should use as the likelihood for quantile regression or how to write a Bayesian model for quantile regression. But it turns out the answer is quite simple, we just need to use the asymmetric Laplace distribution. This distribution has one parameter controlling the mean, another for the scale and a third one for the asymmetry. There are at least two alternative parametrization regarding this asymmetric parameter. In terms of $\kappa$ a parameter that goes from 0 to $\infty$ and in terms of $q$ a number between 0 and 1. This later parametrization is more intuitive for quantile regression as we can directly interpret it as the quantile of interest.
+
+On the next cell we compute the pdf of 3 distribution from the Asymmetric Laplace family
+
+```{code-cell} ipython3
+x = np.linspace(-6, 6, 2000)
+for q, m in zip([0.2, 0.5, 0.8], [0, 0, -1]):
+    κ = (q / (1 - q)) ** 0.5
+    plt.plot(x, stats.laplace_asymmetric(κ, m, 1).pdf(x), label=f"q={q:}, μ={m}, σ=1")
+plt.yticks([])
+plt.legend();
+```
+
+We are going to use a simple dataset to model the Body Mass Index for Dutch kids and young men as a function of their age.
+
+```{code-cell} ipython3
+try:
+    bmi = pd.read_csv(Path("..", "data", "bmi.csv"))
+except FileNotFoundError:
+    bmi = np.loadtxt(pm.get_data("bmi.csv"))
```

Comment:
I think this won't work, np.loadtxt will return and array and the next line `bmi.plot` will fail.